ascat/read_native/nc.py

```diff
@@ -0,0 +1,344 @@
+        looks for all available images on a day on the harddisk. This string is used in
+        datetime.datetime.strftime and in glob.glob to search for all files on a day.
+    file_search_str: string, optional
+        this string is used in datetime.datetime.strftime and glob.glob to find
+        a 3 minute bufr file by the exact date.
+    datetime_format: string, optional
+        datetime format by which {datetime} will be replaced in file_search_str
+    nc_variables: list, optional
+        list of variables to read from netCDF.
+        Default: read all available variables
+    """
+
+    def __init__(self, path, month_path_str='',
+                 day_search_str='W_XX-EUMETSAT-Darmstadt,SURFACE+SATELLITE,METOPA+ASCAT_C_EUMP_%Y%m%d*_125_ssm_l2.nc',
+                 file_search_str='W_XX-EUMETSAT-Darmstadt,SURFACE+SATELLITE,METOPA+ASCAT_C_EUMP_{datetime}*_125_ssm_l2.nc',
+                 datetime_format='%Y%m%d%H%M%S',
+                 filename_datetime_format=(62, 76, '%Y%m%d%H%M%S'),
+                 nc_variables=None):
+        self.path = path
+        self.month_path_str = month_path_str
+        self.day_search_str = day_search_str
+        self.file_search_str = file_search_str
+        self.filename_datetime_format = filename_datetime_format
+        super(AscatL2SsmNc, self).__init__(path, AscatL2SsmNcFile,
+                                           subpath_templ=[month_path_str],
+                                           fname_templ=file_search_str,
+                                           datetime_format=datetime_format,
+                                           exact_templ=False,
+                                           ioclass_kws={
+                                               'nc_variables': nc_variables})
+
+    def _get_orbit_start_date(self, filename):
+        orbit_start_str = \
+            os.path.basename(filename)[self.filename_datetime_format[0]:
+                                       self.filename_datetime_format[1]]
+        return datetime.strptime(orbit_start_str,
+                                 self.filename_datetime_format[2])
+
+    def tstamps_for_daterange(self, startdate, enddate):
+        """
+        Get the timestamps as datetime array that are possible for the
+        given day, if the timestamps are
+
+        For this product it is not fixed but has to be looked up from
+        the hard disk since bufr files are not regular spaced and only
+        europe is in this product. For a global product a 3 minute
+        spacing could be used as a fist approximation
+
+        Parameters
+        ----------
+        start_date : datetime.date or datetime.datetime
+            start date
+        end_date : datetime.date or datetime.datetime
+            end date
+
+        Returns
+        -------
+        dates : list
+            list of datetimes
+        """
+        file_list = []
```

Comment:
I might be wrong but this part seems to be repeated often. Could we make a Baseclass or a Mixin that implements this if it is really identical?

Reply:
I have reviewed the whole code yet, but I agree that a base class makes sense if there is a lot of overlap. @mazweg could you look into that please

Reply:
yes I am looking at it atm and it looks like a base class would indeed make sense.